### PR TITLE
.NET 2.0 slave: initial

### PR DESCRIPTION
**Pull request by tmds**

### CODE REVIEW

File: slave-dotnet20/Dockerfile.rhel7 [hunk 1]
```diff
@@ -0,0 +1,37 @@
+FROM registry.access.redhat.com/openshift3/jenkins-slave-base-rhel7
+
+# Labels consumed by Red Hat build service
+LABEL com.redhat.component="jenkins-slave-dotnet20-rhel7-docker" \
```
Comment (jerboaa) [inline slave-dotnet20/Dockerfile.rhel7]: `rh-dotnet20-jenkins-slave-docker` is the right component.

File: slave-dotnet20/Dockerfile.rhel7 [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      name="openshift3/jenkins-slave-dotnet20-rhel7" \
```
Comment (jerboaa) [inline slave-dotnet20/Dockerfile.rhel7]: `dotnet/dotnet-20-jenkins-slave-rhel7` is the correct name.

File: slave-dotnet20/Dockerfile.rhel7 [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      version="3.6" \
+      architecture="x86_64" \
+      release="1" \
+      io.k8s.display-name="Jenkins Slave .NET Core 2.0" \
```
Comment (jerboaa) [inline slave-dotnet20/Dockerfile.rhel7]: Perhaps ".NET Core 2.0 Jenkins Slave"?

File: slave-dotnet20/Dockerfile.rhel7 [hunk 4]
```diff
@@ -0,0 +1,37 @@
+FROM registry.access.redhat.com/openshift3/jenkins-slave-base-rhel7
+
+# Labels consumed by Red Hat build service
+LABEL com.redhat.component="jenkins-slave-dotnet20-rhel7-docker" \
+      name="openshift3/jenkins-slave-dotnet20-rhel7" \
+      version="3.6" \
```
Comment (jerboaa) [inline slave-dotnet20/Dockerfile.rhel7]: I would go with version of dotnet. I.e. 2.0

File: slave-dotnet20/Dockerfile.rhel7 [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      architecture="x86_64" \
+      release="1" \
+      io.k8s.display-name="Jenkins Slave .NET Core 2.0" \
+      io.k8s.description="The jenkins slave dotnet image has the dotnet tools on top of the jenkins slave base image." \
```
Comment (jerboaa) [inline slave-dotnet20/Dockerfile.rhel7]: Suggested wording: "The .NET Core Jenkins Slave image contains dotnet tools on top of the Jenkins slave base image."

File: slave-dotnet20/Dockerfile.rhel7 [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    ENV=/usr/local/bin/scl_enable \
+    PROMPT_COMMAND=". /usr/local/bin/scl_enable" \
+    ENABLED_COLLECTIONS="rh-nodejs6 rh-dotnet20" \
+    NUGET_XMLDOC_MODE=skip \
+    DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
+
+COPY contrib/bin/scl_enable /usr/local/bin/scl_enable
+
+# Install
+RUN INSTALL_PKGS="$ENABLED_COLLECTIONS" && \
```
Comment (jerboaa) [inline slave-dotnet20/Dockerfile.rhel7]: Would we want to install nss_wrapper? Will there be a problem with resolving usernames with random IDs? Should we install the full `rh-nodejs6` collection or just `rh-nodejs6-npm`? We use the latter for our main builder image.

File: slave-dotnet20/test.sh [hunk 7]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash -e
+#
+# Test the Jenkins image.
+#
+# IMAGE_NAME specifies the name of the candidate image used for testing.
+# The image has to be available before this script is executed.
+#
+
+IMAGE_NAME=${IMAGE_NAME:-openshift3/jenkins-slave-dotnet20-rhel7}
```
Comment (jerboaa) [inline slave-dotnet20/test.sh]: `dotnet/dotnet-20-jenkins-slave-rhel7`

File: build.sh [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    info "Building ${image}"
+    docker rmi -f "${image}" >/dev/null 2>&1 || true
+    docker build -f "${dockerfile}" -t "${image}" "${context_dir}"
+    info "Finished building ${image}"
+}
+
+for v in ${VERSIONS}; do
+    v_no_dot=$(echo ${v} | sed 's/\.//g')
+    slave_dir="${script_dir}/slave-dotnet${v_no_dot}"
+    image="openshift3/jenkins-slave-dotnet${v_no_dot}-rhel7"
```
Comment (jerboaa) [inline build.sh]: `dotnet/dotnet-${v_no_dot}-jenkins-slave-rhel7`

File: build.sh [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+    info "Building ${image}"
+    docker rmi -f "${image}" >/dev/null 2>&1 || true
+    docker build -f "${dockerfile}" -t "${image}" "${context_dir}"
+    info "Finished building ${image}"
+}
+
+for v in ${VERSIONS}; do
+    v_no_dot=$(echo ${v} | sed 's/\.//g')
+    slave_dir="${script_dir}/slave-dotnet${v_no_dot}"
```
Comment (jerboaa) [inline build.sh]: How about having a similar structure than s2i-dotnetcore? I.e. `2.0` subfolder for 2.0 jenkins slave?

**Review by jerboaa — CHANGES_REQUESTED** [review]
Thanks for doing this! A couple of comments below:

**Comment by tmds** [comment]
@jerboaa overview of the labels

slave-maven:
```
LABEL com.redhat.component="jenkins-slave-maven-rhel7-docker" \
      name="openshift3/jenkins-slave-maven-rhel7" \
      version="3.6" \
      architecture="x86_64" \
      release="4" \
      io.k8s.display-name="Jenkins Slave Maven" \
      io.k8s.description="The jenkins slave maven image has the maven tools on top of the jenkins slave base image." \
      io.openshift.tags="openshift,jenkins,slave,maven"
```
slave-nodejs:
```
LABEL com.redhat.component="jenkins-slave-nodejs-rhel7-docker" \
      name="openshift3/jenkins-slave-nodejs-rhel7" \
      version="3.6" \
      architecture="x86_64" \
      release="4" \
      io.k8s.display-name="Jenkins Slave Nodejs" \
      io.k8s.description="The jenkins slave nodejs image has the nodejs tools on top of the jenkins slave base image." \
      io.openshift.tags="openshift,jenkins,slave,nodejs"
```
currently:
```
LABEL com.redhat.component="jenkins-slave-dotnet20-rhel7-docker" \
      name="openshift3/jenkins-slave-dotnet20-rhel7" \
      version="3.6" \
      architecture="x86_64" \
      release="1" \
      io.k8s.display-name="Jenkins Slave .NET Core 2.0" \
      io.k8s.description="The jenkins slave dotnet image has the dotnet tools on top of the jenkins slave base image." \
      io.openshift.tags="openshift,jenkins,slave,dotnet,dotnet20"
```
requested:
```
LABEL com.redhat.component="rh-dotnet20-jenkins-slave-docker" \
      name="dotnet/dotnet-20-jenkins-slave-rhel7" \
      version="2.0" \
      architecture="x86_64" \
      release="1" \
      io.k8s.display-name=".NET Core 2.0 Jenkins Slave" \
      io.k8s.description="The .NET Core Jenkins Slave image contains dotnet tools on top of the Jenkins slave base image." \
      io.openshift.tags="openshift,jenkins,slave,dotnet,dotnet20"
```

I'm going to keep the display-name and description as they are so they are similar to the other slaves when they show up in UI. Will update the other labels as requested.

File: slave-dotnet20/Dockerfile.rhel7 (hunk 6 above)
Comment (tmds) [reply slave-dotnet20/Dockerfile.rhel7]: We want the nodejs-npm package as the s2i builder. nss_wrapper question came up on the openshift/jenkins PR and is not needed in the jenkins slave images.

File: build.sh (hunk 9 above)
Comment (tmds) [reply build.sh]: This is the openshift/jenkins repo naming convention. Will update to match s2i-dotnetcore.

**Review by jerboaa — APPROVED** [review]
Looks good to me.

**Comment by tmds** [comment]
Thanks @jerboaa

**Pull request merged**